relay-event-schema/src/protocol/attributes.rs

```diff
@@ -159,3 +159,82 @@ impl IntoValue for AttributeType {
         serde::ser::Serialize::serialize(self.as_str(), s)
     }
 }
+
+/// Wrapper struct around a collection of attributes with some convenience methods.
+#[derive(Debug, Clone, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+pub struct Attributes(Object<Attribute>);
+
+impl Attributes {
+    /// Creates an empty collection of attributes.
+    pub fn new() -> Self {
+        Self(Object::new())
+    }
+
+    /// Returns the value of the attribute with the given key.
+    pub fn get_value<Q>(&self, key: &Q) -> Option<&Value>
+    where
+        String: Borrow<Q>,
+        Q: Ord + ?Sized,
+    {
+        self.get_attribute(key)?.value.value.value()
+    }
+
+    /// Returns the attribute with the given key.
+    pub fn get_attribute<Q>(&self, key: &Q) -> Option<&Attribute>
+    where
+        String: Borrow<Q>,
+        Q: Ord + ?Sized,
+    {
+        self.0.get(key)?.value()
+    }
+
+    /// Inserts an attribute with the given value into this collection.
+    pub fn insert<V: Into<AttributeValue>>(&mut self, key: String, value: V) {
+        fn inner(slf: &mut Attributes, key: String, value: AttributeValue) {
+            let attribute = Annotated::new(Attribute {
+                value,
+                other: Default::default(),
+            });
+            slf.insert_raw(key, attribute);
+        }
```

Comment:
Why the inner function?

Reply:
(Possibly premature) optimization. It's a trick you can use when you have a function that does nothing with its generic argument except convert it; the inner function doesn't need to be monomorphized.